### PR TITLE
Tidy up docs

### PR DESCRIPTION
**Pull request by diogob**

### CODE REVIEW

File: docs/architecture-notes.md [hunk 1]
```diff
@@ -1,6 +1,36 @@
...
+* jobs - code that runs only on server, either on a clock or upon receiving asynchronous messages.
+
+Besides the section *Definitions* from the [README](../README.md) the framework is agnostic in relation to how the application and its database are modelled. The developer should be able to chose something closer to an event sourced application with immutable data or something more along the lines of mutable entities.
+
+Our main goal is to remove the friction of having to think in terms of HTTP (which is an implementation detail in most systems) or any other protocol and allow the developer to focus on writing the domain logic, which is usually the main concern of the software development process.
+
+Another project goal is to keep interfaces easy to design and enforced on the type level. This should help the developer to code against well defined interfaces reducing bugs and cognitive burden.
+
+### Guiding Principles
```
Comment (diogob) [inline docs/architecture-notes.md:18]: @gustavoguichard special attention to this section. What are your thoughts? Would you remove any of these?

Comment (gustavoguichard) [reply docs/architecture-notes.md:18]: Do we really need to keep the "map of paths not taken"? I wouldn't.. but don't mind if you feel like.
I recon we're probably gonna leave a lot more footprint from here on

Comment (diogob) [reply docs/architecture-notes.md:18]: I would like to preserve the tortuous path that got us here. I'll move to another file so we declutter this one

Comment (diogob) [reply docs/architecture-notes.md:18]: But what do you think of the guiding principles?

Comment (diogob) [reply docs/architecture-notes.md:18]: I have extracted the paths not taken to a separate file simplifying the architecture notes and highlighting the path taken!
I have also added short explanation on why we are keeping that document around and why each alternate architecture was abandoned.

Comment (gustavoguichard) [reply docs/architecture-notes.md:18]: Awesome! I loved the principles, just no really clear to me what is vertical vs horizontal scale (kind of monorepo vs microservices?)

Thanks for extracting out the "paths not taken".

Comment (diogob) [reply docs/architecture-notes.md:18]: yes, vertical scaling is on the same runtime (or server) while horizontal scaling is when you increase throughput but distributing the runtime.

**Pull request merged**